nixos/modules/services/web-apps/netbox.nix

```
@@ -0,0 +1,191 @@
+{ config, lib, pkgs, buildEnv, ... }:
+
+with lib;
+
+let
+  cfg = config.services.netbox;
+  staticDir = cfg.dataDir + "/static";
+  configFile = pkgs.writeTextFile {
+    name = "configuration.py";
+    text = ''
+      STATIC_ROOT = '${staticDir}'
+      ALLOWED_HOSTS = ['*']
+      DATABASE = {
+        'NAME': 'netbox',
+        'USER': 'netbox',
+        'HOST': '/run/postgresql',
+      }
+
+      # Redis database settings. Redis is used for caching and for queuing background tasks such as webhook events. A separate
+      # configuration exists for each. Full connection details are required in both sections, and it is strongly recommended
+      # to use two separate database IDs.
+      REDIS = {
+          'tasks': {
+              'HOST': 'localhost',
+              'PORT': 6379,
+              # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
+              # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
+              # 'SENTINEL_SERVICE': 'netbox',
+              'DATABASE': 0,
+              'SSL': False,
+          },
+          'caching': {
+              'HOST': 'localhost',
+              'PORT': 6379,
+              # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
+              # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
+              # 'SENTINEL_SERVICE': 'netbox',
+              'DATABASE': 1,
+              'SSL': False,
+          }
+      }
+
+      with open("${cfg.secretKeyFile}", "r") as file:
+          SECRET_KEY = file.readline()
+
+      ${cfg.extraConfig}
+    '';
+  };
+  pkg = (cfg.package.overrideAttrs (old: {
+    installPhase = old.installPhase + ''
+      ln -s ${configFile} $out/opt/netbox/netbox/netbox/configuration.py
+    '';
+  })).override {
+    plugins = cfg.plugins;
+  };
+
+in {
+  options.services.netbox = {
+    enable = mkEnableOption "Netbox";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::1]";
+      description = ''
+        Address the server will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8001;
+      description = ''
+        Port the server will listen on.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.netbox;
+      defaultText = literalExpression "pkgs.netbox";
+      description = ''
+        Netbox package to use.
+      '';
+    };
+
+    plugins = mkOption {
+      type = types.listOf types.package;
+      default = [];
+      description = ''
+        Python-Packages to install as Plugins
+      '';
+    };
```

```suggestion
    plugins = mkOption {
      type = types.functionTo (types.listOf types.package);
      default = _: [];
      defaultText = literalExpression ''
        python3Packages: with python3Packages; [];
      '';
      description = ''
        Python-Packages to install as Plugins
      '';
    };
```

And allow a function to be passed, that will be rendered into a list of packages.